index.bs

```diff
@@ -391,6 +391,46 @@ Implementers of this API are expected to be familiar with the <a href="https://g
 
 
 # Programming Model # {#programming-model}
+## Overview ## {#programming-model-overview}
+
+At the heart of neural networks is a computational graph of mathematical operations.
+These operations are the building blocks of modern machine learning technologies in
+computer vision, natural language processing, and robotics.
+The WebNN API is a specification for constructing, compiling, and executing computational
+graphs of neural networks.
+
+The {{MLGraph}} interface represents a compiled computational graph (that is, a model) and exposes
+a compute method to perform inference.
+
+The {{MLGraphBuilder}} interface serves as a builder (factory) to create a {{MLGraph}}.
+The {{MLOperand}} interface is a static (compile-time) representation of runtime values,
+which include input-values for inference, constants (including trained weights)
+used for inference, intermediate values (often referred to as activations) computed
+during inference, as well as the output values of inference.
+The {{MLGraphBuilder}} interface enables the creation of {{MLOperand}}s.
+A key part of the {{MLGraphBuilder}} interface are the operations (such as 
+{{MLGraphBuilder/gemm()}} and {{MLGraphBuilder/softmax()}}). The operations have a functional
+semantics, with no side effects.
+Each operation invocation conceptually returns a distinct new value, without
+changing the value of any other {{MLOperand}}.
+
+The {{MLGraphBuilder/build()}} method of the {{MLGraphBuilder}} interface is used to compile and optimize
+the computation graph used to compute one or more specified outputs. The key
+purpose of the compilation step is to enable optimizations that span two or
+more operations, such as operation or loop fusion.
+
+The runtime values (of {{MLOperand}}s) are tensors, which are essentially multidimensional
+arrays. The representation of the tensors is implementation dependent, but it typically
+includes the array data stored in some buffer (memory) and some metadata describing the
+array data (such as its shape). 
```

Comment:
Similar to the feedback above, I think what missing here is the bind step. For example we can explain it as "before the compiled graph is executed, the actual tensors, which are multidimensional arrays of binary data, need to be bound to the operands in the graph...", etc.

Reply:
Addressed, thanks